src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -707,16 +728,18 @@ class TakeByAggregator[T](var f: (Any) => Any, var n: Int)(implicit var tord: Or
     agg2._state.foreach { case (x, p) => seqOp(x, p) }
   }
 
-  def copy() = new TakeByAggregator(f, n)
+  def copy() = new TakeByAggregator(t, f, n)
 
   private def writeObject(oos: ObjectOutputStream) {
+    oos.writeObject(t)
```

Comment:
FYI, writeObjects and readObjects aren't currently used

Reply:
Those are necessary and implement the serializer interface.